### DISCUSSION TITLE
Group Vertices and Faces by Material Regions

### DISCUSSION THREAD

**Discussion by raducal** [donmccurdy/glTF-Transform#1432] [Q&A] [answered]
I have a model with a single material that wraps it. I'd like to take this model and group vertices and faces together based on the material region. If we take the attached model as an example, the final result should be split into several groups, such as a group (or multiple groups) for the hat, for the cape etc.

I've attempted a few different approaches, but the model always ends up distorted or incorrectly segmented. Does anyone have any suggestions or methods for achieving this accurately?

[model100.glb.zip](https://github.com/user-attachments/files/15754396/model100.glb.zip)

**Comment by donmccurdy** [comment, marked as answer]
@raducal I believe the family of techniques you're describing are usually called "segmentation." This might refer either to splitting regions of a mesh ("mesh segmentation") or regions of an image ("image segmentation") based on some semantic or learned understanding of the scene or image.

You can find papers on these topics, and some implementations like [MeshCNN](https://ranahanocka.github.io/MeshCNN/) and [tensorflow](https://www.tensorflow.org/tutorials/images/segmentation), but these techniques are currently outside the scope of glTF Transform.
